### PR TITLE
Updated ruby version to 3.1.

### PR DESCRIPTION
**Pull request by sreepuramsudheer**
### Description

[Please describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: .expeditor/verify.pipeline.yml [hunk 1]
```diff
@@ -18,14 +18,14 @@ steps:
   # Lint Tests
 #########################################################################
 
-- label: license-scout
```
Comment (marcparadise) [inline .expeditor/verify.pipeline.yml:21]: We can't safely merge without resolving license scout issues.

Comment (jashaik) [reply .expeditor/verify.pipeline.yml:21]: @sreepuramsudheer Please revert this change.

**Comment by jashaik** [comment]
https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/6899

File: .expeditor/automate_build.sh [hunk 2]
```diff
@@ -42,6 +42,7 @@ HAB_FEAT_OFFLINE_INSTALL=true HAB_FEAT_IGNORE_LOCAL=false HAB_ORIGIN=chef HAB_CA
 
 git clone https://github.com/chef/automate.git
 cd automate
+git checkout ssudheer/ruby_update_3.1
```
Comment (jashaik) [inline .expeditor/automate_build.sh:45]: Is this a temporary change? Can we make this configurable? Future work ?

File: .expeditor/chef_server.sh [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 git clone https://github.com/chef/automate.git
 cd automate
-git checkout kalroy/cs_plan_changes
+git checkout ssudheer/ruby_update_3.1
```
Comment (jashaik) [inline .expeditor/chef_server.sh:9]: Make it configurable but for now it's ok.

File: .license_scout.yml [hunk 4]
```diff
@@ -213,4 +213,6 @@ exceptions:
       reason: Exception made by Chef Legal
     - name: core/nss-myhostname
       reason: Exception made by Chef Legal
+    - name: core/xz
+      reason: Added for test to pass
```
Comment (jashaik) [inline .license_scout.yml:217]: Update the comment accordingly.

**Review by jashaik — APPROVED** [review]

**Review by talktovikas — APPROVED** [review]

File: .expeditor/chef_server.sh (hunk 3 above)
Comment (sreepuramsudheer) [reply .expeditor/chef_server.sh]: Made it configurable.

File: .expeditor/automate_build.sh (hunk 2 above)
Comment (sreepuramsudheer) [reply .expeditor/automate_build.sh]: Made it configurable.

**Pull request merged**